### PR TITLE
[New] Untrusted DLL Loaded by Azure AD Sync Service

### PR DESCRIPTION
**Pull request by Samirbous**
converted an existing ER diag rule to SIEM rule to also answer a detection request in our community slack channel https://elasticstack.slack.com/archives/C016E72DWDS/p1728897639096869 

added https://medium.com/@breakingmhet/detect-azure-pass-through-authentication-abuse-azure-hybrid-environments-ed4274784252 also to the references.

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: New - Guidelines

These guidelines serve as a reminder set of considerations when proposing a new rule.

### Documentation and Context

- [ ] Detailed description of the rule.
- [ ] List any new fields required in ECS/data sources.
- [ ] Link related issues or PRs.
- [ ] Include references.

### Rule Metadata Checks

- [ ] `creation_date` matches the date of creation PR initially merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive, considering performance for diverse environments. Non ecs fields should be added to `non-ecs-schema.json` if not available in an integration.
- [ ] `min_stack_comments` and `min_stack_version` should be included if the rule is only compatible starting from a specific stack version.
- [ ] `index` pattern should be neither too specific nor too vague, ensuring it accurately matches the relevant data stream (e.g., use logs-endpoint.process-* for process data).
- [ ] `integration` should align with the `index`. If the integration is newly introduced, ensure the manifest, schemas, and `new_rule.yaml` template are updated.
- [ ] `setup` should include the necessary steps to configure the integration.
- [ ] `note` should include any additional information (e.g. Triage and analysis investigation guides, timeline templates).
- [ ] `tags` should be relevant to the threat and align/added to the `EXPECTED_RULE_TAGS` in the definitions.py file.
- [ ] `threat`, `techniques`, and `subtechniques` should map to ATT&CK always if possible.

#### New BBR Rules
- [ ] `building_block_type` should be included if the rule is a building block and the rule should be located in the `rules_building_block` folder.
- [ ] `bypass_bbr_timing` should be included if adding custom lookback timing to the rule.

### Testing and Validation

- [ ] Provide evidence of testing and detecting the expected threat.
- [ ] Check for existence of coverage to prevent duplication.

File: rules/windows/credential_access_imageload_azureadconnectauthsvc.toml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+[metadata]
+creation_date = "2024/10/14"
+integration = ["endpoint", "windows"]
+maturity = "production"
+min_stack_comments = "Breaking change at 8.13.0 for SentinelOne Integration."
+min_stack_version = "8.13.0"
```
Comment (w0rk3r) [inline rules/windows/credential_access_imageload_azureadconnectauthsvc.toml:6]:
```suggestion
```

File: rules/windows/credential_access_imageload_azureadconnectauthsvc.toml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+updated_date = "2024/10/14"
+
+[rule]
+author = ["Elastic", "Matteo Potito Giorgio"]
+description = """
+Identifies the load of a library with no or invalid code signature and by the Azure AD Sync process, which may 
+indicate an attempt to persist or collect sensitive credentials passing through the Azure AD synchronization server.
```
Comment (w0rk3r) [inline rules/windows/credential_access_imageload_azureadconnectauthsvc.toml:13]:
```suggestion
Identifies the load of a DLL without a valid code signature by the Azure AD Sync process, which may indicate an attempt
to persist or collect sensitive credentials passing through the Azure AD synchronization server.
```
Optional

File: rules/windows/credential_access_imageload_azureadconnectauthsvc.toml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Credential Access", "Data Source: Elastic Defend", "Data Source: Sysmon"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+any where host.os.type == "windows" and process.name : "AzureADConnectAuthenticationAgentService.exe" and 
+(
+ (event.category == "library" and event.action == "load") or 
+ (event.category == "process" and event.action : "Image loaded*")
+ ) and
```
Comment (w0rk3r) [inline rules/windows/credential_access_imageload_azureadconnectauthsvc.toml:37]:
```suggestion
) and 
```
nit

**Review by w0rk3r — APPROVED** [review]

File: rules/windows/credential_access_imageload_azureadconnectauthsvc.toml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          "?:\\Windows\\WinSxS\\*",
+          "?:\\Windows\\System32\\DriverStore\\FileRepository\\*") or 
+          
+   /* Sysmon DLL path is mapped to file.path */
+   file.path :
+         ("?:\\Windows\\assembly\\NativeImages*",
+          "?:\\Windows\\Microsoft.NET\\*",
+          "?:\\Windows\\WinSxS\\*",
+          "?:\\Windows\\System32\\DriverStore\\FileRepository\\*")
+  )
```
Comment (Aegrah) [inline rules/windows/credential_access_imageload_azureadconnectauthsvc.toml:55]:
```suggestion
any where host.os.type == "windows" and process.name like "AzureADConnectAuthenticationAgentService.exe" and 
(
 (event.category == "library" and event.action == "load") or 
 (event.category == "process" and event.action : "Image loaded*")
 ) and 

not (?dll.code_signature.trusted == true or file.code_signature.status == "Valid") and not 

  (
   /* Elastic defend DLL path */
   ?dll.path like
         ("?:\\Windows\\assembly\\NativeImages*",
          "?:\\Windows\\Microsoft.NET\\*",
          "?:\\Windows\\WinSxS\\*",
          "?:\\Windows\\System32\\DriverStore\\FileRepository\\*") or 
          
   /* Sysmon DLL path is mapped to file.path */
   file.path like
         ("?:\\Windows\\assembly\\NativeImages*",
          "?:\\Windows\\Microsoft.NET\\*",
          "?:\\Windows\\WinSxS\\*",
          "?:\\Windows\\System32\\DriverStore\\FileRepository\\*")
  )
```

Comment (Samirbous) [reply rules/windows/credential_access_imageload_azureadconnectauthsvc.toml:55]: on windows process.name and dll.path are case sensitive.

**Pull request merged**